src/phylum/init/cli.py

```diff
@@ -45,23 +45,61 @@
 }
 
 TOKEN_ENVVAR_NAME = "PHYLUM_TOKEN"
-PHYLUM_PATH = pathlib.Path.home() / ".phylum"
-PHYLUM_BIN_PATH = PHYLUM_PATH / "phylum"
-SETTINGS_YAML_PATH = PHYLUM_PATH / "settings.yaml"
 
-# Potential features to add:
-#  * Add logging support, a verbosity option to control it, and swap out print statements for logging
-#  * Check for valid versions by using the GitHub API to compare against actual releases
-#    * If so, also programmatically generate the SUPPORTED_TARGET_TRIPLES
-#  * Use the GitHub API ("https://api.github.com") to more programmatically get the archive URL
-#  * Add a `--list` option, to show which versions are available
-#  * Add an option to account for pre-releases
+
+def get_phylum_settings_path(version):
+    """Get the Phylum settings path based on a provided version."""
+    home_dir = pathlib.Path.home()
+    version = version_check(version)
+
+    config_home_path = os.getenv("XDG_CONFIG_HOME")
+    if not config_home_path:
+        config_home_path = home_dir / ".config"
+    phylum_config_path = pathlib.Path(config_home_path) / "phylum" / "settings.yaml"
+
+    # The Phylum config path changed following the v2.2.0 release, to adhere to the XDG Base Directory Spec
+    # Reference: https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html
+    if Version(canonicalize_version(version)) <= Version("v2.2.0"):
+        phylum_config_path = home_dir / ".phylum" / "settings.yaml"
+
+    return phylum_config_path
+
+
+def get_phylum_bin_path(version):
+    """Get the path to the Phylum binary based on a provided version."""
+    home_dir = pathlib.Path.home()
+    version = version_check(version)
+
+    # The Phylum binary path changed following the v2.2.0 release, to adhere to the XDG Base Directory Spec
+    # Reference: https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html
+    phylum_bin_path = home_dir / ".local" / "bin" / "phylum"
+
+    if Version(canonicalize_version(version)) <= Version("v2.2.0"):
+        phylum_bin_path = home_dir / ".phylum" / "phylum"
+
+    return phylum_bin_path
+
+
+def get_latest_version():
+    """Get the "latest" version programmatically and return it."""
+    # API Reference: https://docs.github.com/en/rest/releases/releases#get-the-latest-release
+    github_api_url = "https://api.github.com/repos/phylum-dev/cli/releases/latest"
+
+    req = requests.get(github_api_url, timeout=5.0)
+    req.raise_for_status()
+    req_json = req.json()
+
+    # The "name" entry stores the GitHub Release name, which could be set to something other than the version.
+    # Using the "tag_name" entry is better since the tags are much more tightly coupled with the release version.
+    latest_version = req_json.get("tag_name")
+
+    return latest_version
 
 
 def version_check(version):
     """Check a given version for validity and return a normalized form of it."""
     if version == "latest":
-        return version
+        version = get_latest_version()
```

Comment:
This should make the `if version == "latest"` condition in `get_archive_url()` obsolete, right?

Reply:
Correct. The latest commit has the change to reflect this.